datafusion/src/datasource/object_store/local.rs

```diff
@@ -39,19 +38,22 @@ pub struct LocalFileSystem;
 
 #[async_trait]
 impl ObjectStore for LocalFileSystem {
-    async fn list_file(&self, prefix: &str) -> Result<FileMetaStream> {
+    async fn list_file(&self, prefix: &str) -> Result<FileMetaStream, GenericError> {
```

Comment:
this is fine to me, though I also think it is fine to keep this returning a `DataFusion` error too as it is straightforward to construct them from `Box<..>` errors. 🤔